### DISCUSSION TITLE
Github Pages

### DISCUSSION THREAD

**Discussion by SalehMoradi-art** [community/community#60403] [Repositories] [answered]
### Select Topic Area

Question

### Body

Hi , i have a problem with publishing a web on Github

for example i have a repository and i want to publish it but it wont be on my github domain
it is like this :
https://salehmoradi-art.github.io/Page/

and i want it directly go to index.html by this domain like this :
https://salehmoradi-art.github.io

**Comment by GhostOf0days** [comment]
Deploy with [Netlify](https://www.netlify.com/blog/2016/09/29/a-step-by-step-guide-deploying-on-netlify/) instead. It takes less than 5 minutes. I would avoid Github Pages. Make sure for Netlify, you have a _redirects file with '/* /index.html 200' in it. Instead of Github Pages, use Netlify, Heroku, or Vercel. Use Netlify for simple use cases/websites and Vercel for larger, more complex ones.

**Reply by GhostOf0days** [reply]
Due to the _redirects file, it would not only be better to deploy on Netlify, but it will go to index.html. Above '/* /index.html 200', put '/* /:splat  200.'

**Reply by GhostOf0days** [reply]
Also, instead of an href like 'location.href='https://salehmoradi-art.github.io/Page/port.html', please only put the file name from the [relative path](https://www.geeksengine.com/article/absolute-relative-path.html#:~:text=Relative%20path%20is%20a%20path,form%20of%20a%20short%20version.) if they are in the same repo. For instance, use './port.html' here instead of the website stuff before it.

**Comment by melvincwng** [comment, marked as answer]
Hi @SalehMoradi-art, thanks for posting your query on the GitHub Community discussions.

I've took a look at your repositories over here: https://github.com/SalehMoradi-art?tab=repositories, and noted that you have 2 repositories named - Page and salehmoradi.githib.io.

![image](https://github.com/community/community/assets/77479885/fb13730d-a843-4f7b-95e6-d4f3b2869d8a)

The reason why this occurs:
> for example i have a repository and i want to publish it but it wont be on my github domain
it is like this :
https://salehmoradi-art.github.io/Page/

Is because you are deploying your Page repository using GitHub Pages.


If instead, you wish for this to occur - i.e.:
> and i want it directly go to index.html by this domain like this :
https://salehmoradi-art.github.io/

You need to create a repository called **salehmoradi-art.github.io**, add your static files over there, and deploy it again using GitHub Pages. Then your website will appear in this format - i.e. https://salehmoradi-art.github.io/. Just FYI, there could be a typo in the naming of your second repository (salehmoradi.githib.io), not sure if that's intended.

Following the instructions above should resolve your issue.

Hope this helps and answers your question, have a good day ahead 😄

**Reply by SalehMoradi-art** [reply]
Hi thank you for the information it worked , I really appreciate it

**Reply by melvincwng** [reply]
Pleasure to help! ❤️
